tests/tests_json_database.py

```diff
@@ -33,104 +32,128 @@ def test_database_add_many(tmpdir):
-        db.update({"name": "test"}, {"name": "test works!"})
+        a.updateByQuery({"name": "test"}, {"name": "test works!"})
     with pytest.raises(SchemaError):
-        db.update({"naame": "test"}, {"namer": "test works!"})
+        a.updateByQuery({"naame": "test"}, {"namer": "test works!"})
 
 
 def test_database_update_by_id(tmpdir):
     file = tmpdir.join("test.db.json")
     file.write(ID_FIXTURE_STR)
-    db = Database().on(file.strpath, uuid=False)
-    db.updateById(ID_FIXTURE["data"][0]["id"], {"name": "test works!"})
-    x = db.get()[0]["name"]
-    assert db.get()[0]["name"] == "test works!"
+    a = getDb(file.strpath)
+    a.updateById(ID_FIXTURE["data"][0]["id"], {"name": "test works!"})
+    x = a.get()[0]["name"]
+    assert a.get()[0]["name"] == "test works!"
 
 
 def test_database_update_by_id_not_found(tmpdir):
     file = tmpdir.join("test.db.json")
     file.write(ID_FIXTURE_STR)
-    db = Database().on(file.strpath, uuid=False)
+    a = getDb(file.strpath)
     with pytest.raises(IdNotFoundError):
-        db.updateById(123, {"name": "not found :("})
+        a.updateById(123, {"name": "not found :("})
 
 
 def test_database_delete_by_id(tmpdir):
     file = tmpdir.join("test.db.json")
     file.write(ID_FIXTURE_STR)
-    db = Database().on(file.strpath, uuid=False)
-    assert db.deleteById(ID_FIXTURE["data"][0]["id"])
-    assert not bool(len(db.get()))
+    a = getDb(file.strpath)
+    assert a.deleteById(ID_FIXTURE["data"][0]["id"])
+    assert bool(len(a.get())) is True
     fixture = [
         {"name": "test", "getbyfield": "row1"},
         {"name": "test works!", "getbyfield": "row2"},
         {"name": "testing, so much fun .. yeah.", "getbyfield": "row3"},
     ]
-    db.addMany(fixture)
-    assert len(db.getAll()) == 3
-    assert db.deleteById(db.get()[0]["id"])
-    assert len(db.getAll()) == 2
-    assert db.deleteById(db.get()[0]["id"])
-    assert len(db.getAll()) == 1
-    assert db.deleteById(db.get()[0]["id"])
-    assert not bool(len(db.get()))
+
+    a.addMany(fixture)
+    assert len(a.getAll()) == 3
+    assert a.deleteById(a.get()[0]["id"])
+    assert len(a.getAll()) == 2
+    assert a.deleteById(a.get()[0]["id"])
+    assert len(a.getAll()) == 1
+    assert a.deleteById(a.get()[0]["id"])
+    assert bool(len(a.get())) is True
     with pytest.raises(IdNotFoundError):
-        assert db.deleteById(20)
+        assert a.deleteById(20)
+
+
+def test_database_delete_all(tmpdir):
+    file = tmpdir.join("test.db.json")
+    file.write(ID_FIXTURE)
+    a = getDb(file.strpath)
+    assert a.deleteAll() is None
+
+
+def test_database_research(tmpdir):
+    file = tmpdir.join("test.db.json")
+    file.write(EMPTY_FIXTURE_STR)
+    a = getDb(file.strpath)
+    fixture = [
+        {"name": "test", "getbyfield": "row1"},
+        {"name": "pysondb", "getbyfield": "row2"},
+        {"name": "def23@c-py", "getbyfield": "row3"},
+        {"name": "stuff(py", "type": "GUI"}
```

Comment:
`{"name": "test", "getbyfield": "row1"},` this line sets the schema of the DB to have two keys, that are, name and getByField. so when you add the last one `{"name": "stuff(py", "type": "GUI"}` this no longer compiles to the schema and thus is not added to the DB. and thats why you are getting an IndexError

Reply:
Thanks brother